maze_dataset/utils.py

```diff
@@ -235,3 +235,13 @@ def timeit_fancy(
         return_value=return_value,
         profile=profile,
     )
+
+
+def unpackable_if_true_attribute(
+    iterable: Iterable, attr_owner: Any, attr_name: str
+) -> Iterable:
+    """Returns `iterable` if `attr_owner` has the attribute `attr_name` and it boolean casts to `True`.
+    Particularly useful for optionally inserting delimiters into a sequence depending on an `TokenizerElement` attribute.
+    What a terrible name, please help.
```

Comment:
maybe `attribute_gated_sequence`? I'd also clarify in the docstring that it returns an empty iterable if the attribute is `False` or missing